base/strings/unicode.jl

```diff
@@ -634,8 +634,8 @@ graphemes(s::AbstractString) = GraphemeIterator{typeof(s)}(s)
 eltype(::Type{GraphemeIterator{S}}) where {S} = SubString{S}
 eltype(::Type{GraphemeIterator{SubString{S}}}) where {S} = SubString{S}
 
-function length(g::GraphemeIterator)
-    c0 = typemax(Char)
+function length(g::GraphemeIterator{S}) where {S}
+    c0 = eltype(S)(0x00000000)
```

Comment:
Is `'\0'` a safe character for this? It's not a  malformed character – does the Unicode spec ensure that it can't ever combine with anything?

Reply:
Yes.

Reply:
(I used `\0` here since hopefully every conceivable `AbstractChar` type can encode this.)